planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -102,6 +102,35 @@ double project_goal_to_map(
   return project.z();
 }
 
+geometry_msgs::msg::Pose get_closest_centerline_pose(
+  const lanelet::ConstLanelets & road_lanelets, const geometry_msgs::msg::Pose & point,
+  vehicle_info_util::VehicleInfo vehicle_info)
+{
+  lanelet::Lanelet closest_lanelet;
+  lanelet::utils::query::getClosestLanelet(road_lanelets, point, &closest_lanelet);
+
+  const auto refined_center_line = lanelet::utils::generateFineCenterline(closest_lanelet, 1.0);
+  closest_lanelet.setCenterline(refined_center_line);
+
+  const double lane_yaw = lanelet::utils::getLaneletAngle(closest_lanelet, point.position);
+
+  const auto nearest_idx =
+    motion_utils::findNearestIndex(convertCenterlineToPoints(closest_lanelet), point.position);
+  const auto nearest_point = closest_lanelet.centerline()[nearest_idx];
+
+  // update y coordinate according to the asymmetric vehicle margin
```

Comment:
Maybe sounds better?

```suggestion
  // shift nearest point on its local y axis so that vehicle's right and left edges would have approx the same clearance from road border 
```

Reply:
done :+1: